lib/smack/Regions.cpp

```diff
@@ -143,15 +101,22 @@ void Region::init(const Value *V, unsigned length) {
   }
 
   singleton =
-      DL && representative && isSingleton(representative, offset, length);
+      DL && representative && isSingleton(V, length);
   allocated = !representative || isAllocated(representative);
+  // Shaobo: I removed the dependency of the bytewise flag on isFieldDisjoint
+  // and isMemcpyd for the following reasons.
+  // TODO: figure out if removing isFieldDisjoint breaks value annotations.
+  // Since we visit memcpy/memset instructions in this pass, the types
+  // of the nodes involving in these instructions are either i8 or changed to
+  // NULL during merging.
   bytewise = DSA && SmackOptions::BitPrecise &&
              (SmackOptions::NoByteAccessInference ||
-              !isFieldDisjoint(DSA, V, offset) ||
-              DSA->isMemcpyd(representative) || T->isIntegerTy(8));
-  incomplete = !representative || representative->isIncompleteNode();
+              !DSA->isTypeSafe(V) ||
+              //DSA->isMemcpyd(representative) ||
+             T->isIntegerTy(8));
+  incomplete = !representative || representative->isIncomplete();
   complicated = !representative || isComplicated(representative);
-  collapsed = !representative || representative->isCollapsedNode();
+  collapsed = !representative || representative->isOffsetCollapsed();
```

Comment:
Do you understand the difference between offset-collapsed and type-collapsed?
Why are you using offset-collapsed here and not type-collapsed?

Reply:
No, I don't understand the difference. In our email communications with @caballa, he mentioned that `isOffsetCollapsed` is equivalent to LLVM DSA's `isCollapsedNode`. So I simply used it without thinking further more about the difference.

Reply:
It would be good to understand the difference. So please ask them about this or learn it from their code.

Reply:
Any news on this one?

Reply:
> Any news on this one?

I think Arie and his student replied to us. The bottom line here is that we're using the correct flag here.